### ISSUE TITLE
Missing Implementor and Abstraction in Bridge Pattern

### ISSUE THREAD

**Issue by sheiun** [faif/python-patterns#295]
The code <https://github.com/faif/python-patterns/blob/master/patterns/structural/bridge.py>
different from the source <https://en.wikibooks.org/wiki/Computer_Science_Design_Patterns/Bridge#Python>

The Bridge Pattern's UML

![image](https://user-images.githubusercontent.com/36428458/56626415-2b49b100-6674-11e9-9a90-ce16eac55ac7.png)

**Comment by gyermolenko** [comment]
hi @sheiun , thanks for the issue.

1. In terms of the pattern: Abstraction/interface delegates the work to the implementation/platform.
It is not about interfaces or abstract classes in any particular programming language.

2. As I see it - it is not always necessary to create python ABC classes to emulate something like interface from other languages.

TLDR: don't see a problem here.

Thoughts?

**Comment by faif** [comment]
One of the benefits of duck typing is that it doesn't force you to define interfaces for everything. You can still do it if you want with a module like `abc` as mentioned by @gyermolenko, but it's not mandatory. I also find the current implementation fine.
